src/controller/CommissioneeDeviceProxy.h

```diff
@@ -0,0 +1,392 @@
+
+    bool IsSecureConnected() const override { return IsActive() && mState == ConnectionState::SecureConnected; }
+
+    bool IsSessionSetupInProgress() const { return IsActive() && mState == ConnectionState::Connecting; }
+
+    void Reset();
+
+    NodeId GetDeviceId() const override { return mDeviceId; }
+
+    bool MatchesSession(SessionHandle session) const { return mSecureSession.HasValue() && mSecureSession.Value() == session; }
+
+    chip::Optional<SessionHandle> GetSecureSession() const override { return mSecureSession; }
+
+    Messaging::ExchangeManager * GetExchangeManager() const override { return mExchangeMgr; }
+
+    void SetAddress(const Inet::IPAddress & deviceAddr) { mDeviceAddress.SetIPAddress(deviceAddr); }
+
+    PASESession & GetPairing() { return mPairing; }
+
+    uint8_t GetNextSequenceNumber() override { return mSequenceNumber++; };
+
+    CHIP_ERROR LoadSecureSessionParametersIfNeeded()
+    {
+        bool loadedSecureSession = false;
+        return LoadSecureSessionParametersIfNeeded(loadedSecureSession);
+    };
+
+    CHIP_ERROR SetCSRNonce(ByteSpan csrNonce)
+    {
+        VerifyOrReturnError(csrNonce.size() == sizeof(mCSRNonce), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mCSRNonce, csrNonce.data(), csrNonce.size());
+        return CHIP_NO_ERROR;
+    }
+
+    ByteSpan GetCSRNonce() const { return ByteSpan(mCSRNonce, sizeof(mCSRNonce)); }
+
+    CHIP_ERROR SetAttestationNonce(ByteSpan attestationNonce)
+    {
+        VerifyOrReturnError(attestationNonce.size() == sizeof(mAttestationNonce), CHIP_ERROR_INVALID_ARGUMENT);
+        memcpy(mAttestationNonce, attestationNonce.data(), attestationNonce.size());
+        return CHIP_NO_ERROR;
+    }
+
+    ByteSpan GetAttestationNonce() const { return ByteSpan(mAttestationNonce, sizeof(mAttestationNonce)); }
+
+    bool AreCredentialsAvailable() const { return (mDAC != nullptr && mDACLen != 0); }
+
+    ByteSpan GetDAC() const { return ByteSpan(mDAC, mDACLen); }
+    ByteSpan GetPAI() const { return ByteSpan(mPAI, mPAILen); }
+
+    CHIP_ERROR SetDAC(const ByteSpan & dac);
+    CHIP_ERROR SetPAI(const ByteSpan & pai);
+
+    MutableByteSpan GetMutableNOCCert() { return MutableByteSpan(mNOCCertBuffer, sizeof(mNOCCertBuffer)); }
+
+    CHIP_ERROR SetNOCCertBufferSize(size_t new_size);
+
+    ByteSpan GetNOCCert() const { return ByteSpan(mNOCCertBuffer, mNOCCertBufferSize); }
+
+    MutableByteSpan GetMutableICACert() { return MutableByteSpan(mICACertBuffer, sizeof(mICACertBuffer)); }
+
+    CHIP_ERROR SetICACertBufferSize(size_t new_size);
+
+    ByteSpan GetICACert() const { return ByteSpan(mICACertBuffer, mICACertBufferSize); }
+
+    Controller::DeviceControllerInteractionModelDelegate * GetInteractionModelDelegate() override { return mpIMDelegate; };
+
+private:
+    enum class ConnectionState
+    {
+        NotConnected,
+        Connecting,
+        SecureConnected,
+    };
+
+    enum class ResetTransport
+    {
+        kYes,
+        kNo,
+    };
```

Comment:
I think ResetTransport is no longer used and can be removed.

Reply:
Let me see if it can be removed here in this PR itself. Otherwise, it'll be removed for sure when https://github.com/project-chip/connectedhomeip/pull/10808/files#diff-8f9de4143e87a3aad6e7fce591bf052bd08e83a037737cfd6f57f61c3ee8e3e5R46 is handled.